infra/modules/doc-upload-s3/main.tf

```diff
@@ -0,0 +1,165 @@
+  bucket = aws_s3_bucket.doc-upload.id
+  policy = data.aws_iam_policy_document.doc-upload.json
+}
+
+
+resource "aws_s3_bucket_versioning" "doc-upload" {
+  bucket = aws_s3_bucket.doc-upload.id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "doc-upload" {
+  bucket = aws_s3_bucket.doc-upload.bucket
+
+  rule {
+    apply_server_side_encryption_by_default {
+      kms_master_key_id = aws_kms_key.doc-upload.arn
+      sse_algorithm     = "aws:kms"
+    }
+  }
+}
+
+resource "aws_kms_key" "doc-upload" {
+  description = "KMS key for ${var.environment_name} Document Upload"
+  # The waiting period, specified in number of days. After the waiting period ends, AWS KMS deletes the KMS key.
+  deletion_window_in_days = "10"
+  # Generates new cryptographic material every 365 days, this is used to encrypt your data. The KMS key retains the old material for decryption purposes.
+  enable_key_rotation = "true"
+}
+
+# ----------------------------------------------------
+#
+#  Document Upload Logging
+#
+# ----------------------------------------------------
+
+# Create the S3 bucket to provide server access logging.
+resource "aws_s3_bucket" "doc-upload-log" {
+  bucket = "${var.environment_name}-upload-logging"
+
+  # checkov:skip=CKV_AWS_144:Cross region replication not required by default
+  # checkov:skip=CKV2_AWS_61:This S3 bucket should not transition objects to another storage class or expire objects
+  # checkov:skip=CKV2_AWS_62:Disable SNS requirement
+}
+resource "aws_s3_bucket_logging" "doc-upload-log" {
+  bucket        = aws_s3_bucket.doc-upload.id
+  target_bucket = aws_s3_bucket.doc-upload-log.bucket
+  target_prefix = var.environment_name
+}
+
+resource "aws_s3_bucket_versioning" "doc-upload-log" {
+  bucket = aws_s3_bucket.doc-upload-log.id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "doc-upload-log" {
+  bucket = aws_s3_bucket.doc-upload-log.bucket
+
+  rule {
+    apply_server_side_encryption_by_default {
+      kms_master_key_id = aws_kms_key.doc-upload.arn
+      sse_algorithm     = "aws:kms"
+    }
+  }
+}
+
+resource "aws_s3_bucket_public_access_block" "doc-upload-log" {
+  bucket = aws_s3_bucket.doc-upload-log.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+resource "aws_s3_bucket_policy" "doc-upload-log" {
+  bucket = aws_s3_bucket.doc-upload-log.id
+  policy = data.aws_iam_policy_document.doc-upload-log.json
+}
+# ----------------------------------------------------
+#
+#  Data Attributes
+#
+# ----------------------------------------------------
+
+data "aws_iam_role" "task_executor" {
+  # Referencing the task executor of the ECS services so that they have the ability to upload documents to s3
+  name = "${var.service_name}-task-executor"
+
+}
```

Comment:
Instead of looking up a single task executor, we can make this module more flexible by:
1. Creating customer managed IAM policies with associated policy documents
2. Passing in the role arns as a variable that should have these policies attached to them
3. Calling this s3 module _after_ creating the services in the env-template and then passing in the service task executor arns

That way, the module's job is just:
- create and configure the s3 bucket
- create the kms key
- create roles to access the s3 bucket

The env-template is the module that knows how to wire everything up together.